front/app/modules/commercial/user_custom_fields/hooks/useUserJsonFormSchemas.ts

```diff
@@ -22,19 +22,28 @@ export default function useUserJsonFormsSchemas() {
   useEffect(() => {
     const subscription = userJsonFormSchemasStream().observable.subscribe(
       (customFields) => {
-        setCustomFields({
-          ...customFields,
-          hasRequiredFields:
-            !isNilOrError(locale) &&
-            !isEmpty(
-              get(customFields, `json_schema_multiloc.${locale}.required`, null)
-            ),
-          hasCustomFields:
-            !isNilOrError(locale) &&
-            !customFields?.ui_schema_multiloc?.[locale]?.elements.every(
-              (e) => e?.options?.hidden
-            ),
-        });
+        setCustomFields(
+          !isNilOrError(locale)
+            ? {
+                schema: customFields.json_schema_multiloc[locale],
+                uiSchema: customFields.ui_schema_multiloc[locale],
+                hasRequiredFields:
+                  !isNilOrError(locale) &&
+                  !isEmpty(
+                    get(
+                      customFields,
+                      `json_schema_multiloc.${locale}.required`,
+                      null
+                    )
+                  ),
+                hasCustomFields:
+                  !isNilOrError(locale) &&
+                  !customFields?.ui_schema_multiloc?.[locale]?.elements.every(
+                    (e) => e?.options?.hidden
+                  ),
```

Comment:
I think I removed the `hidden` option from the json_forms_service in the BE, and just took it out of the schema completely, so we can probably drop this.